src/tree.rs

```diff
@@ -436,6 +438,63 @@ impl<T> Tree<T> {
     }
 
     ///
+    /// Swaps two `Node`s including their children given their `NodeId`s.
+    ///
+    /// Returns an empty `Result` containing a `NodeIdError` if one occurred on either provided `Id`.
+    ///
+    /// Notes:
+    ///     - Both `NodeId`s are still valid after this process and are not swapped.
+    ///     - Keeps the positions of the `Node`s in their parents children collection.
+    ///
+    /// ``
```

Comment:
On mobile, so I'm sorry for any typos.

I totally agree with you that we need a brand new (from scratch) solution to this, but I'm not really sure what it should be. 

My gut reaction is that it would be neat if we could just swap the data in the nodes and swap their children (the entire Vec), but then the NodeIds would be backwards right?  I'm not 100% sure, but wouldn't swapping the two Nodes in self.nodes fix that?